### PR TITLE
[4078] Add listening for events guide

### LINKED ISSUE #4078: Write a guide on how to listen for events
### Context & proposal

Create a guide in the common guides section on how to subscribe for events.

Also, as an example we can showcase how `NotificationRemovedFromChannelEvent` event can be used to close the chat screen when the current user is removed from the channel.

```kotlin
override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
     // ...

     ChatClient.instance().subscribeFor<NotificationRemovedFromChannelEvent>(viewLifecycleOwner) {
            val removedUserId = it.member.user.id
            val currentUserId = ChatClient.instance().clientState.user.value?.id

            if (removedUserId == currentUserId) {
                // Close the current chat screen as the current user has been removed.
                requireActivity().onBackPressed()
            }
     }
}
```

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Create "Listening for events" guide in the common section
- [ ] Add a snippet that shows how to subscribe/unsubscribe to/from events
- [ ] Add a section that shows how we can use events to close the chat screen

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

Closes #4078 

Add a guide that demonstrates how use events to close the chat screen when the current user has been removed from the channel.

### 🎨 UI Changes

screencapture-localhost-3000-chat-docs-sdk-android-client-guides-listening-for-events-2022-10-26-17_40_19

### 🧪 Testing

1. Run the docs by executing `npx stream-chat-docusaurus -i -s`
2. Navigate to "Listening for Events" guide in the common section

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

**Comment by adasiewiczr** [comment]
Can you add Java snippets as well?

File: docusaurus/docs/Android/02-client/06-guides/11-listening-for-events.mdx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+val channelClient = client.channel("messaging", "general")
+
+// Subscribe for new message events
+val disposable: Disposable = channelClient.subscribeFor<NewMessageEvent> { newMessageEvent ->
+    val message = newMessageEvent.message
+}
+
+// Dispose when you want to stop receiving events
+disposable.dispose()
+```
```
Comment (skydoves) [inline docusaurus/docs/Android/02-client/06-guides/11-listening-for-events.mdx:15]: Just for curiosity, when do we call the `dispose` method typically in the lifecycle? (onPause or onDestroy)

**Review by skydoves — APPROVED** [review]
Overall looks good to me!

File: docusaurus/docs/Android/02-client/06-guides/11-listening-for-events.mdx (hunk 1 above)
Comment (bychkovdmitry) [reply docusaurus/docs/Android/02-client/06-guides/11-listening-for-events.mdx:22]: That's actually a very good question. If the subscription was done in `onViewCreated`, then I would do dispose in `onDestroyView`.

Comment (skydoves) [reply docusaurus/docs/Android/02-client/06-guides/11-listening-for-events.mdx:22]: Thanks! I'm wondering if I can work on auto-dispose features based on the Android lifecycle for the subscription.

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**